openwillis/measures/audio/speech_phonation.py

```diff
@@ -0,0 +1,214 @@
+
+    except Exception as e:
+        logger.error(f'Error phonation extraction: {e} & File: {filepath}')
+
+    return phonation_dict
+
+def clean_acoustic_df(df, file, duration, measures):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    This function cleans the acoustic dataframe.
+
+    Parameters:
+    ...........
+    df : pandas.DataFrame
+        input dataframe
+    file : str
+        file name
+    duration : float
+        duration of the audio signal in seconds
+    measures : dict
+        column names configuration
+
+    Returns:
+    ...........
+    pandas.DataFrame
+        cleaned dataframe
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    df[measures['phonation_type']] = file.split('_')[-1][0]
+    df = df[[measures['phonation_type']] + [col for col in df.columns if col != measures['phonation_type']]]
+    df[measures['duration']] = duration
+
+    # remove unused columns - pause related measures
+    cols_to_drop = [measures['spir'], measures['pause_meddur'], measures['pause_maddur'], measures['silence_ratio']]
+    df = df.drop(columns=[col for col in cols_to_drop if col in df.columns], errors='ignore')
+
+    return df
+
+def phonations_acoustics(audio_path, transcript_json, speaker_label=''):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    This function extracts phonation acoustics from the audio signal based on the transcripted information,
+     then computes advanced vocal acoustics measures.
+
+    Parameters:
+    ...........
+    audio_path : str
+        Path to the input audio file.
+    transcript_json : json
+        Speech transcription json response.
+    speaker_label : str
+        Speaker label.
+
+    Returns:
+    ...........
+    phonations_df : pandas.DataFrame
+        A pandas dataframe containing the phonation-level measures.
+    summ_df : pandas.DataFrame
+        A pandas dataframe containing the summarized phonation acoustics measures.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+
+    phonations_df, summ_df = pd.DataFrame(), pd.DataFrame()
+    measure_path = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), './config/acoustic.json'))
+    file = open(measure_path)
+    measures = json.load(file)
+
+    try:
+        if audio_path.split('.')[-1] != 'wav':
+            raise Exception('Audio file format not supported. Please provide a .wav file.')
+
+        # extract phonation segments
+        phonation_dict = phonation_extraction(audio_path, transcript_json, speaker_label)
+        if not phonation_dict:
+            return phonations_df, summ_df
+
+        # save phonation segments
+        temp_dir = tempfile.mkdtemp()
```

Comment:
just checking: do we need to pass the path ? where will it create the temp folder?

I just want to make sure we do not delete any other item when we delete this directory.

Reply:
This will be created in the default temp directory of the system I guess. Since this is generated from this function it shouldn't interfere with any other files we deal with. Deleting it should be safe, since the only files that will exist inside it are the extracted phonation audio files